internal/services/appservice/service_plan_data_source.go

```diff
@@ -131,6 +131,7 @@ func (r ServicePlanDataSource) Read() sdk.ResourceFunc {
 			}
 
 			servicePlan.Location = location.NormalizeNilable(existing.Location)
+			servicePlan.Kind = *existing.Kind
```

Comment:
To avoid a potential crash, this should be:
```suggestion
			servicePlan.Kind = utils.NormalizeNilableString(existing.Kind)
```